content/blog/multitenancy-for-kubernetes-clusters-using-hpe-alletra-6000-and-nimble-storage.md

```diff
@@ -0,0 +1,234 @@
+
+Visit [HPE Storage Container Orchestrator Documentation](https://scod.hpedev.io) (SCOD) to learn more about what storage resources are being exposed to tenants.
+
+# Example use cases
+
+There are plenty of different use cases that multitenancy enables for IT Ops looking to manage and secure storage resources for a diverse set of applications running on Kubernetes.
+
+## Ephemeral Inline Volumes
+
+End users that deploy applications on Kubernetes that require ephemeral storage at a capacity beyond of what a worker node is capable of providing, may use Ephemeral Inline Volumes. Before multitenancy, Kubernetes administrators had to share the `Secret` with the end user to allow provisioning of the Ephemeral Inline Volume. That is not very practical from a security standpoint as application administrators would have privileges on the storage array. Now, the end user may request a separate tenant to allow management of Ephemeral Inline Volumes securely for their application.
+
+Example.
+
+```yaml
+kind: Pod
+apiVersion: v1
+metadata:
+  name: my-csi-app-inline-volume
+spec:
+  containers:
+    - name: my-frontend
+      image: busybox
+      command: [ "sleep", "100000" ]
+      volumeMounts:
+      - mountPath: "/data"
+        name: my-csi-volume
+  volumes:
+  - name: my-csi-volume
+    csi:
+      driver: csi.hpe.com
+      nodePublishSecretRef:
+        name: my-tenant-secret
+      fsType: ext3
+      volumeAttributes:
+        csi.storage.k8s.io/ephemeral: "true"
+        accessProtocol: "iscsi"
+        size: "5Gi"
+```
+
+The `Secret` "my-tenant-secret" would have to exist in the same `Namespace` as the `Pod` and contain the necessary connectivity and credentials to the tenant.
+
+## Virtualization and Containers
+
+Deploying Kubernetes on a virtualization platform such as VMware vSphere, OpenStack or Hyper-V is by far the most popular pattern for deploying on-premises Kubernetes. Many times customers want to leverage the same array to provide persistent storage both for the virtualization platform and the container platform.
+
+Allowing Kubernetes clusters "administrator" or "poweruser" access to the array served by the virtualization platform the cluster is running on might be feasible in a single tenant and single application type scenario. Once weaving in the Ephemeral Inline Volumes use case into the mix and we've basically given application administrators way too many privileges on the array.
+
+In many cases the virtualization and storage administrator is combined into the same role, moving forward, this administrative function would be able to securely hand over credentials to Kubernetes administrators that needs a first class persistent storage solution.
```

Comment:
Place a period after "...the same role." Start the next sentence with "Moving forward, this administrative..."
Change "needs a first class" to "need a first class"